applications/StructuralMechanicsApplication/tests/cpp_tests/test_metric_process.cpp

```diff
@@ -0,0 +1,351 @@
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometries/triangle_2d_3.h"
+#include "geometries/tetrahedra_3d_4.h"
+#include "testing/testing.h"
+#include "includes/kratos_flags.h"
+#include "includes/gid_io.h"
+
+/* Processes */
+#include "processes/compute_nodal_gradient_process.h"
+#include "custom_processes/metrics_spr_error_process.h"
+
+namespace Kratos 
+{
+    namespace Testing 
+    {
+        typedef Node<3> NodeType;
+      
+        void GiDIODebugMetricSPR(ModelPart& ThisModelPart)
+        {
+            GidIO<> gid_io("TEST_METRIC_SPR", GiD_PostBinary, SingleFile, WriteUndeformed,  WriteElementsOnly);
+            const int nl_iter = ThisModelPart.GetProcessInfo()[NL_ITERATION_NUMBER];
+            const double label = static_cast<double>(nl_iter);
+
+            gid_io.InitializeMesh(label);
+            gid_io.WriteMesh(ThisModelPart.GetMesh());
+            gid_io.FinalizeMesh();
+            gid_io.InitializeResults(label, ThisModelPart.GetMesh());
+            gid_io.WriteNodalResults(DISPLACEMENT, ThisModelPart.Nodes(), label, 0);
+            gid_io.PrintOnGaussPoints(ERROR_INTEGRATION_POINT, ThisModelPart, label);
+            gid_io.PrintOnGaussPoints(CAUCHY_STRESS_VECTOR, ThisModelPart, label);
+            gid_io.PrintOnGaussPoints(STRAIN_ENERGY, ThisModelPart, label);
+            const bool meshing_application_compiled = KratosComponents<Variable<Vector>>::Has("MMG_METRIC");
+            KRATOS_WARNING_IF("CalculateMetric", !meshing_application_compiled) << "Please compile the MeshingApplication in order to compute the metric" << std::endl;
+            const Variable<Vector>& metric_variable = KratosComponents<Variable<Vector>>::Get("MMG_METRIC");
+            if (meshing_application_compiled)
+                gid_io.WriteNodalResultsNonHistorical(metric_variable, ThisModelPart.Nodes(), label);
+        }
+        
+        void Create2DGeometry(ModelPart& ThisModelPart, const std::string& ElementName)
+        {
+            Properties::Pointer p_elem_prop = ThisModelPart.pGetProperties(0);
+
+            // First we create the nodes
+            NodeType::Pointer p_node_1 = ThisModelPart.CreateNewNode(1, 0.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_2 = ThisModelPart.CreateNewNode(2, 1.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_3 = ThisModelPart.CreateNewNode(3, 1.0 , 1.0 , 0.0);
+            NodeType::Pointer p_node_4 = ThisModelPart.CreateNewNode(4, 0.0 , 1.0 , 0.0);
+            NodeType::Pointer p_node_5 = ThisModelPart.CreateNewNode(5, 2.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_6 = ThisModelPart.CreateNewNode(6, 2.0 , 1.0 , 0.0);
+
+            // Now we create the "conditions"
+            std::vector<NodeType::Pointer> element_nodes_0 (3);
```

Comment:
Better would be to use one line initializer lists. Also, these lines appear to get copied between many test files. Maybe these can be cleaned up in a separate PR.